src/log.ts

```diff
@@ -143,6 +143,10 @@ export namespace message {
   }
 
   // SCALE
+  export const CANNOT_UNION_CUSTOM_DOMAIN_WITH_FIELD_DOMAIN = 'custom domain scale cannot be unioned with default field-based domain';
+
+  export const CANNOT_USE_INTERPOLATE_WITH_NON_COLOR = 'Cannot use scheme with non-color channel.';
```

Comment:
That said, why don't you make it a function, so you make the code in the rule shorter?